src/build/vite/plugin.ts

```diff
@@ -124,13 +126,50 @@ export async function nitro(
         builder: {
           /// Share the config instance among environments to align with the behavior of dev server
           sharedConfigBuild: true,
-          async buildApp(builder) {
-            await buildProduction(ctx, builder);
-          },
         },
       };
     },
 
+    buildApp: {
+      order: "post",
+      handler(builder) {
+        return buildProduction(ctx, builder);
+      },
+    },
+
+    generateBundle: {
+      handler(_options, bundle) {
+        const { root } = this.environment.config;
+        // find entry point of this service
+        let entryFile: string | undefined;
+        for (const [_name, file] of Object.entries(bundle)) {
+          if (file.type === "chunk") {
+            if (file.isEntry) {
+              if (entryFile !== undefined) {
+                this.error(
+                  `Multiple entry points found for service "${this.environment.name}". Only one entry point is allowed.`
```

Comment:
Maybe we only show warning if this.environment is server consumer (or more specifically a defined servicr/ssr)